profiling/src/internal/observation/observations.rs

```diff
@@ -235,9 +210,9 @@ mod test {
         };
 
         let mut o = Observations::default();
-        o.add(s1, None, vec![1, 2, 3]);
+        o.add(s1, None, vec![1, 2, 3]).unwrap();
         // This should panic
-        o.add(s2, None, vec![4, 5]);
+        o.add(s2, None, vec![4, 5]).unwrap();
```

Comment:
Now that we are returning an error, we can remove the `should_panic` on line 199, and confirm that its always an error using:
```suggestion
        o.add(s2, None, vec![4, 5]).unwrap_err();
```

Reply:
Nice catch, the tests were actually passing for the wrong reason. Fixed in 8bc38208ab43ffcc9116d59b9c23ca74100239cb